File: 1003-check-if-word-is-valid-after-substitutions/1003-check-if-word-is-valid-after-substitutions.java
class Solution {
    public boolean isValid(String s) {
        List<Character> list = new ArrayList<>();
        for(char c : s.toCharArray()) {
            list.add(c);
            int n = list.size();
            if(n >= 3 && list.get(n-3) == 'a' && list.get(n-2) == 'b' && list.get(n-1) == 'c') {
                list.remove(n-1);
                list.remove(n-2);
                list.remove(n-3);
            }
        }
        return list.size() == 0;
    }
}

File: 1003-check-if-word-is-valid-after-substitutions/1003-check-if-word-is-valid-after-substitutions.cpp
class Solution {
public:
    bool isValid(string s) {
        vector<char> chars;
        for(char c : s) {
            chars.push_back(c);
            int n = chars.size();
            if(n >= 3) {
                if(chars[n-3] == 'a' && chars[n-2] == 'b' && chars[n-1] == 'c') {
                    chars.pop_back(); chars.pop_back(); chars.pop_back();
                }
            }
        }
        return chars.size() == 0;
    }
};